skylines/frontend/views/flight.py

```diff
@@ -448,26 +448,39 @@ def update():
     if 'pilot_id' in data:
         pilot_id = data['pilot_id']
 
-        if pilot_id is not None and not User.exists(id=pilot_id):
-            return jsonify(error='unknown-pilot'), 422
+        if pilot_id is not None:
 
-        if g.flight.pilot_id != pilot_id:
-            g.flight.pilot_id = pilot_id
+            if not User.exists(id=pilot_id):
+                return jsonify(error='unknown-pilot'), 422
 
-            # update club if pilot changed
-            if pilot_id is not None:
+            if g.flight.pilot_id != pilot_id and User.get(pilot_id).club_id == g.current_user.club_id:
+                g.flight.pilot_id = pilot_id
+                # pilot_name is irrelevant, if pilot_id is given
+                g.flight.pilot_name = None
+                # update club if pilot changed
                 g.flight.club_id = User.get(pilot_id).club_id
```

Comment:
since we're using `User.get(pilot_id).club_id` twice now we should probably extract it into a local variable to prevent the duplicate `get()` call.